### PR TITLE
improve test coverage in operator/pkg/controlplane

### PR DESCRIPTION
**Pull request by adiprerepa**
I had to export some fields, which may be why the diff is large. I closed the previous PR because for some reason, my new commits werent showing up.

### CODE REVIEW

**Comment by adiprerepa** [comment]
@howardjohn I closed the prev. PR because my changes were not showing up, would you review this one?

**Comment by adiprerepa** [comment]
@morvencao

**Review by morvencao — COMMENTED** [review]
/ok-to-test

File: operator/pkg/controlplane/control_plane.go [hunk 1]
```diff
@@ -104,6 +104,30 @@ func (i *IstioOperator) Run() error {
 	return nil
 }
 
+func (i *IstioOperator) ComponentsEqual(components []component.IstioComponent) bool {
+	if i.components == nil && components == nil {
+		return true
+	}
+	if (i.components == nil && components != nil) || (i.components != nil && components == nil) || len(i.components) != len(components) {
+		return false
+	}
```
Comment (morvencao) [inline operator/pkg/controlplane/control_plane.go:113]: why not simplify this to:

```
if (len(i.components) != len(components) {
        return false
}
```

File: operator/pkg/controlplane/control_plane_test.go [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+							Options:       coreComponentOptions,
+							ComponentName: name.IstioBaseComponentName,
+						},
+					},
+					&component.PilotComponent{
+						CommonComponentFields: &component.CommonComponentFields{
+							Options: &component.Options{
+								InstallSpec: &v1alpha1.IstioOperatorSpec{},
+								Translator:  &translate.Translator{},
+							},
```
Comment (morvencao) [inline operator/pkg/controlplane/control_plane_test.go:99]: equal to 
```
Options:       coreComponentOptions,
```
?

Comment (morvencao) [reply operator/pkg/controlplane/control_plane_test.go:99]: In addition, IMO we may need some more complicated cases, constructing IstioOperator object with more complicated Translator, especially gateways and addonComponents, as they are slice and map type.

Comment (adiprerepa) [reply operator/pkg/controlplane/control_plane_test.go]: @morvencao that is a good idea, but I think I will do that in a separate PR. Is everything else OK? I'm still learning how all these components to operator work.

Comment (morvencao) [reply operator/pkg/controlplane/control_plane_test.go]: sure, thanks!

**Review by morvencao — APPROVED** [review]

**Pull request merged**